app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -861,6 +868,11 @@ public function load($printQuery = false, $logQuery = false)
         }
         Varien_Profiler::start('__EAV_COLLECTION_BEFORE_LOAD__');
         Mage::dispatchEvent('eav_collection_abstract_load_before', array('collection' => $this));
+
+        if ($this->_eventPrefix !== 'eav_collection_abstract') {
```

Comment:
What's the reasoning for this?

Reply:
This is to prevent that an event with same name gets fired twice. In this class the "default" event prefix is set to `eav_collection_abstract`. If a custom class extends this it uses this prefix and the new event has not to be fired again.